presto-main/src/test/java/com/facebook/presto/sql/analyzer/TestAnalyzer.java

```diff
@@ -1597,6 +1597,13 @@ public void testJoinUnnest()
         analyze("SELECT * FROM (VALUES array[2, 2]) a(x) FULL OUTER JOIN UNNEST(x) ON true");
     }
 
+    @Test
+    public void testUnnestInnerScalar()
+    {
+        analyze("SELECT * FROM (SELECT array[1,2] a) a CROSS JOIN UNNEST(a) AS T(x)");
+        assertFails(NOT_SUPPORTED, "SELECT * FROM (SELECT 1) CROSS JOIN UNNEST((SELECT array[1])) AS T(x)");
```

Comment:
Also check that the error message has the correct line/col/text using regexp match